pandas/core/dtypes/dtypes.py

```diff
@@ -1132,6 +1134,12 @@ def __new__(cls, subtype=None, closed: str_type | None = None):
                 "for IntervalDtype"
             )
             raise TypeError(msg)
+        elif is_any_real_numeric_dtype(subtype) and not is_64bit_real_numeric_dtype(
```

Comment:
wont this allow through e.g. Int64 that we don't want?  this seems a lot heavier than just checking itemsize